### PR TITLE
Throw IOException if `mkdirs()` fails while unpacking ZIP file (#359)

### PR DESCRIPTION
**Pull request by slovdahl**

### CODE REVIEW

**Comment by slovdahl** [comment]
I'm not sure if it's possible to write a test that triggers the `IOException`. Do you have any ideas @decebals?

**Comment by slovdahl** [comment]
Two ideas:
- Create the plugin directory that the unzip would create, and change the owner/group of it to something that isn't writeable by the current user. But that approach has at least two problems IMO:
  1. The temporary folders created by tests can't be removed.
  1. It makes the test less portable between platforms. I'm unsure if this is a problem in practice, maybe pf4j already is fairly tied to Linux-like environments?
- Create the plugin directory that the unzip would create and a subfolder from the ZIP file, and remove the writeable/executable permission of the subfolder. This didn't work though, as the `Unzip` class does this before extracting:
  ```java
  // delete destination directory if exists
  if (destination.exists() && destination.isDirectory()) {
      FileUtils.delete(destination.toPath());
  }
  ```

File: pf4j/src/main/java/org/pf4j/util/Unzip.java [hunk 1]
```diff
@@ -78,29 +78,34 @@ public void extract() throws IOException {
...
                     }
-                } catch (FileNotFoundException e) {
-                    log.error("File '{}' not found", zipEntry.getName());
                 }
             }
         }
     }
 
+    private static void mkdirsOrThrow(File dir) throws IOException {
+        if (!dir.exists()) {
```
Comment (decebals) [inline pf4j/src/main/java/org/pf4j/util/Unzip.java]: It's not an issue but I think that it's better to put the both checks on a single line (`if (!dir.exists() && !dir.mkdirs())`)

Comment (slovdahl) [reply pf4j/src/main/java/org/pf4j/util/Unzip.java]: Thanks, I don't know what I was thinking here :smile:

Fixed.

**Comment by decebals** [comment]
Thanks!

**Pull request merged**